### ISSUE TITLE
"TypeError: :param data: MUST be bytes." in properties.py

### ISSUE THREAD

**Issue by gdesmar** [TeamMsgExtractor/msg-extractor#357]
**Bug Metadata**
* Version of extract_msg: v0.40.0 or next-release
* Your python version: Python 3.9.15

**Describe the bug**
When opening one of the two messages contained in this [ParamBytes.zip](https://github.com/TeamMsgExtractor/msg-extractor/files/11347196/ParamBytes.zip) zip file, the library is raising an error.
The files are coming from:
https://www.virustotal.com/gui/file/6f4fa43ad0df38a91fb4099b174845c23fabfb30745134736c6eaceda0d05cb4
https://www.virustotal.com/gui/file/6ad28ee3d1fad0b6bf07dca6ba98cf687a0362510ae3e615246f6ce25454076e
You can open the zip file using `TeamMsgExtractor`.
It may be normal to raise an error, but I would assume that one of extract_msg.exceptions would be more fitting to explain the problem behind the parsing.

**Code to reproduce this error**
```python
>>> import extract_msg
>>> extract_msg.openMsg("6ad28ee3d1fad0b6bf07dca6ba98cf687a0362510ae3e615246f6ce25454076e")
```

**Traceback**
```
Traceback (most recent call last):
  File "msg-extractor/extract_msg/msg.py", line 640, in attachments
    return self._attachments
AttributeError: 'Message' object has no attribute '_attachments'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "msg-extractor/extract_msg/utils.py", line 667, in openMsg
    return Message(path, **kwargs)
  File "msg-extractor/extract_msg/message_base.py", line 88, in __init__
    super().__init__(path, **kwargs)
  File "msg-extractor/extract_msg/msg.py", line 162, in __init__
    self.attachments
  File "msg-extractor/extract_msg/msg.py", line 654, in attachments
    self._attachments.append(self.attachmentClass(self, attachmentDir))
  File "msg-extractor/extract_msg/attachment.py", line 35, in __init__
    super().__init__(msg, dir_)
  File "msg-extractor/extract_msg/attachment_base.py", line 34, in __init__
    self.__props = Properties(self._getStream('__properties_version1.0'), PropertiesType.ATTACHMENT)
  File "msg-extractor/extract_msg/properties.py", line 26, in __init__
    raise TypeError(':param data: MUST be bytes.')
TypeError: :param data: MUST be bytes.
```

**Comment by TheElementalOfDestruction** [comment]
Ah, that's just an improperly handled error. It means that your properties stream was missing in the Attachment *entirely* which is a complete violation of the standard ("there is exactly one property steam" is a requirement for attachment object storages. Your file is either corrupt or whatever produced it messed up badly. If outlook produced it, that's a big issue. I'll see about fixing the handling and possibly adding a way to just ignore standards violation errors.

**Comment by TheElementalOfDestruction** [comment]
Alright, next-release should properly raise the correct error now. I'll look into a way to ignore them for people analyzing files later.

**Comment by TheElementalOfDestruction** [comment]
Decided to do some analysis on your files, turns out the files are pretty badly messed up, and I'm not sure why. the 6ad one is the most obviously messed up since it's alignment is wrong (much be a series of 512 byte blocks and it isn't, so some sector was probably created with a block that was either too small or a bit too big). Turning on some of the defect reporting for OleFileIO and then checking it myself revealed that that file *also* points to a block that is more than 1000 blocks outside the scope of the file. Either the creation code from the file is severely faulty or the file was cut off at some point.

The other file *also* point to blocks that are far outside of the file for some reason. Additionally, the reason *it* has an error is because the file tree (or what of it exists) shows that there are no files in the attachment folder.

TL;DR: What on earth made these files, because they violate the CBF format, not just the MSG format?

**Comment by TheElementalOfDestruction** [comment]
Alright, the new code using the `ErrorBehavior` enum and the `errorBehavior` argument to `openMsg` will allow you to open the files and at least moderately read their contents. The 6ad one had a PDF attachment that also appears to have been corrupted in some way. The other file also looked to have a PDF on it, but no valid references to it could be found

**Comment by TheElementalOfDestruction** [comment]
Closing this issue as it appears to be resolved. Let me know if it is not.
